client/controller/localized-links.js

```diff
@@ -4,9 +4,14 @@ import performanceMark from 'calypso/server/lib/performance-mark';
 import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import { setDocumentHeadLink } from 'calypso/state/document-head/actions';
 
-const getLocalizedCanonicalUrl = ( path, locale ) => {
-	const baseUrl = `https://wordpress.com${ path }`;
-	const baseUrlWithoutLang = baseUrl.replace(
+const getLocalizedCanonicalUrl = ( path, locale, excludeSearch = false ) => {
+	const baseUrl = new URL( path, 'https://wordpress.com' );
+
+	if ( excludeSearch ) {
+		baseUrl.search = '';
+	}
```

Comment:
Putting this logic in `getLocalizedCanonicalUrl` means we get the same treatment for both `<link rel=canonical>` and `<link rel=alternate hreflang>`.

Reply:
Maybe we should rename `excludeSearchFromCanonicalUrl()` to better convey that? Something like `excludeSearchFromCanonicalUrlAndHrefLangLinks`?